From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Thu, 5 Dec 2019 16:34:39 +0000
Subject: Revert "staging: bcm2835_camera: Ensure all buffers are returned on
 disable"

This reverts commit d70c57a7c6a069977a0d827e1d5e2de53fb87e6d.

The patch has mutated beyond recognition and created merge errors
against 37c35be "staging: mmal-vchiq: Add support for event callbacks."

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c b/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c
index 341594ab669f..61a3593cdcfc 100644
--- a/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c
+++ b/drivers/staging/vc04_services/vchiq-mmal/mmal-vchiq.c
@@ -268,8 +268,6 @@ static void buffer_work_cb(struct work_struct *work)
 	if (!buffer->cmd)
 		atomic_dec(&msg_context->u.bulk.port->buffers_with_vpu);
 
-	atomic_dec(&msg_context->u.bulk.port->buffers_with_vpu);
-
 	msg_context->u.bulk.port->buffer_cb(msg_context->u.bulk.instance,
 					    msg_context->u.bulk.port,
 					    msg_context->u.bulk.status,
